nengo/utils/stdlib.py

```diff
@@ -138,6 +138,50 @@ def discard(self, key):
             del self._data[key]
 
 
+class FrozenOrderedSet(Set):
+    """A set that preserves insertion order and is hashable."""
+
+    def __init__(self, data=None):
+        if data is None:
+            data = []
+        self.data = dict((d, None) for d in data)
```

Comment:
I would use `OrderedDict` here for now, so that we don't have differences between Python 3.5 and 3.6+.

Reply:
I was thinking we'd merge this after the PR dropping Python 3.5 support, but yeah if not then we should do that.